src/views/LimitOrders/hooks/useGelatoLimitOrdersHistory.ts

```diff
@@ -4,23 +4,42 @@ import useSWR from 'swr'
 import { SLOW_INTERVAL } from 'config/constants'
 import { useMemo } from 'react'
 
-import { getLSOrders, saveOrder } from 'utils/localStorageOrders'
+import { getLSOrders, saveOrder, removeOrder, saveOrders, hashOrderSet, hashOrder } from 'utils/localStorageOrders'
 import useGelatoLimitOrdersLib from 'hooks/limitOrders/useGelatoLimitOrdersLib'
 
-import { ORDER_CATEGORY } from '../types'
+import { ORDER_CATEGORY, LimitOrderType } from '../types'
 
 function newOrdersFirst(a: Order, b: Order) {
   return Number(b.updatedAt) - Number(a.updatedAt)
 }
 
-function syncOrderToLocalStorage({ chainId, account, orders }) {
-  const ordersLS = getLSOrders(chainId, account)
-
-  orders.forEach((order: Order) => {
-    const orderExists = ordersLS.find((confOrder) => confOrder.id.toLowerCase() === order.id.toLowerCase())
-
-    if (!orderExists || (orderExists && Number(orderExists.updatedAt) < Number(order.updatedAt))) {
-      saveOrder(chainId, account, order)
+function syncOrderToLocalStorage({
+  chainId,
+  account,
+  orders,
+  syncTypes,
+}: {
+  chainId: number
+  account: string
+  orders: Order[]
+  syncTypes: LimitOrderType[]
+}) {
+  const allOrdersLS = getLSOrders(chainId, account)
+
+  const allOrdersLSHashSet = hashOrderSet(allOrdersLS)
+  const newOrders = orders.filter((order: Order) => !allOrdersLSHashSet.has(hashOrder(order)))
+  saveOrders(chainId, account, newOrders)
+
+  const typeOrders = orders.filter((order) => syncTypes.some((type) => type === order.status))
+  const otherOrders = orders.filter((order) => syncTypes.some((type) => type !== order.status))
+  const typeOrdersLS = allOrdersLS.filter((order) => syncTypes.some((type) => type === order.status))
+  typeOrdersLS.forEach((confOrder: Order) => {
+    const updatedOrder = typeOrders.find((order) => confOrder.id.toLowerCase() === order.id.toLowerCase())
+
+    if (updatedOrder && Number(confOrder.updatedAt) < Number(updatedOrder.updatedAt)) {
+      saveOrder(chainId, account, updatedOrder)
+    } else if (!updatedOrder && otherOrders.some((order) => confOrder.id.toLowerCase() === order.id.toLowerCase())) {
```

Comment:
If the order is not found desired status but updated in graph, remove it from that type of orders